frontend/discourse/app/controllers/topic.js

```diff
@@ -377,10 +377,61 @@ export default class TopicController extends Controller {
   }
 
   @action
-  editTopic(event) {
+  async handleTitleClick(event) {
+    this.editTopic?.(event);
+    this.jumpTop?.(event);
+  }
+
+  @action
+  async editTopic(event) {
     event?.preventDefault();
-    if (this.get("model.details.can_edit")) {
-      this.set("editingTopic", true);
+    const canEditTitle = this.get("model.details.can_edit");
+    const canLocalize = this.currentUser?.can_localize_content;
+
+    if (!canEditTitle && !canLocalize) {
+      return;
+    }
```

Comment:
Shouldn't this be `||` or checked below before the actions? For the case of i.e. "they can edit titles but can't localize" they should be able to `this.set("editingTopic", true)` or `this.openComposerfForEdit(topic, firstPost)`